e2e/libs/node_exec/node_exec.py

```diff
@@ -3,15 +3,31 @@
 import time
 import logging
 from utility.utility import wait_delete_pod
+from utility.utility import wait_delete_ns
 
 DEFAULT_POD_TIMEOUT = 180
 DEFAULT_POD_INTERVAL = 1
 
 class NodeExec:
 
-    def __init__(self, namespace):
-        self.namespace = namespace
+    _instance = None
+
+    @staticmethod
+    def get_instance():
+        if NodeExec._instance is None:
+            NodeExec()
+        return NodeExec._instance
```

Comment:
This singleton will not be thread safe, but this can be improved later if needed.